concourse/steps/scan_container_images.py

```diff
@@ -84,12 +84,11 @@ def has_results(self):
     def mail_body(self):
         parts = []
         parts.append(self._mail_disclaimer())
-        parts.append(protecode_results_table(
-            protecode_cfg=self._protecode_cfg,
-            upload_results=self._protecode_results,
-            )
-        )
-        parts.append(self._clamav_report())
+
+        if self._protecode_results:
```

Comment:
that's something I did think about (but not yet implement): we need to take into account whether or not a scan has been done.

id est: if both protecode and clamav scans have been done. protecode yielded some results, but no malware was found. then there should be a part in the mail that states about the successful virus scan. OTOH, if only protecode scans have been done, it should _not_ be (falsely) stated that a clamav scan was done. Compare to my remark about applying a factory-ish pattern